xclim/indices/_agro.py

```diff
@@ -1298,7 +1298,7 @@ def standardized_precipitation_evapotranspiration_index(
     """
     fitkwargs = fitkwargs or {}
 
-    dist_methods = {"gamma": ["ML", "APP", "PWM"], "fisk": ["ML", "APP"]}
+    dist_methods = {"gamma": ["ML", "APP"], "fisk": ["ML", "APP"]}
```

Comment:
Why remove "PWM"?  I thought the conclusion was to change the accepted types of `dist` instead, to allow passing `lmoments3` classes ?

Reply:
Right now it's not working, so the smallest change possible was to forbid it for now. 

But anyways, the way I see it, the strings here will correspond to constrained and well tested flagships distributions. The PWM/gamma would use lmoments3 and fall in the category of uncontrolled distributions. Those changes would: 

* In SPI signature: `dist : str -> dist : str | scipy.stats.rv_continuous` 
* Only do checks if `dist` is a str:
```python
    if isinstance(dist,str):  # only check str, which are the flagship distributions/methods
        if dist in dist_methods.keys():
            if method not in dist_methods[dist]:
                raise NotImplementedError(
                    f"{method} method is not implemented for {dist} distribution"
                )
        else:
            raise NotImplementedError(f"{dist} distribution is not yet implemented.")
 ```

Reply:
So, in short, I was thinking I could add this in another PR, and keep this one focused and short and sweet. But I want to remove PWM right now, because it's not working.

Reply:
This strategy is good with me!